packages/rest/src/providers/log-error-provider.ts

```diff
@@ -6,20 +6,18 @@
 import {Provider} from '@loopback/context';
 import {ServerRequest} from 'http';
 import {LogError} from '../internal-types';
-import * as debugModule from 'debug';
-const debug = debugModule('loopback:rest:error');
 
 export class LogErrorProvider implements Provider<LogError> {
```

Comment:
+1 to have a provider.

Reply:
Can we add it as an optional dependency of the `DefaultSequence`?

Reply:
It is already a _required_ dependency of the `DefaultSequence`. Changing it to an _optional_ dependency is out of scope of this pull request IMO. Please note that I am not changing the design of LogError sequence action in this patch, just the implementation details.

Reply:
I don't see it being injected at https://github.com/strongloop/loopback-next/blob/e7c6f888a2f9603c7c0ae5dd8c6a4af274e05030/packages/rest/src/sequence.ts#L61. Can you clarify?

Reply:
Errors are logged by the `reject` action. The LogError function is injected into our default implementation of `reject`, see

https://github.com/strongloop/loopback-next/blob/45a4bdf2ccd0a92e647ff78acea730442d78abde/packages/rest/src/providers/reject.ts#L13-L17

This allows us to keep the sequence as a high-level description of steps to perform, and let individual sequence actions to deal with lower-level concerns like error handling and logging.

https://github.com/strongloop/loopback-next/blob/45a4bdf2ccd0a92e647ff78acea730442d78abde/packages/rest/src/sequence.ts#L105-L114